src/react.ts

```diff
@@ -2,18 +2,17 @@ import { useDebugValue } from 'react'
 import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/shim/with-selector'
 import createStore, {
   EqualityChecker,
-  GetState,
-  SetState,
+  Mutate,
   State,
   StateCreator,
   StateSelector,
   StoreApi,
+  StoreMutatorIdentifier,
 } from './vanilla'
 
-export function useStore<T extends State>(api: StoreApi<T>): T
-export function useStore<T extends State, U>(
-  api: StoreApi<T>,
-  selector: StateSelector<T, U>,
+export function useStore<S extends StoreApi<State>, U = ExtractState<S>>(
+  api: S,
+  selector?: StateSelector<ExtractState<S>, U>,
```

Comment:
Oh, I see. We actually support undefined selector with defined equalityFn.

(I didn't know `StateSelector<ExtractState<S>, U>` is possible with `U=...`. TIL)

Reply:
> We actually support undefined selector with defined equalityFn.

Yes that would work with these types. (Also at the end I'll add `| undefined` to all optional signatures)

> U=...

Haha yeah with that you don't need overloads

Reply:
> Also at the end I'll add `| undefined` to all optional signatures

Why do you need it? I usually don't add it for simplicity.

Reply:
So that it works for people who have `--exactOptionalProperties` on. For example this won't compile if you have it on...

```ts
declare const foo: (a: number, b?: number) => void

foo(1, undefined)
```

Reply:
Hmm, in my mind, that's something I want to have an error, if there's such an option.

exactOptionalProperties only works on properties, and not function parameters? https://tsplay.dev/wQeZJw

Reply:
> Hmm, in my mind, that's something I want to have an error, if there's such an option.

But then passing `undefined` as selector and some value as equalityFn won't work.

> exactOptionalProperties only works on properties, and not function parameters? https://tsplay.dev/wQeZJw

Ah interesting, then there's no point of add `| undefined` to parameters

Reply:
> But then passing `undefined` as selector and some value as equalityFn won't work.

Ah, I see. That makes sense. Anyway, for now, we don't need to add `| undefined`.